examples/vello_editor/src/text.rs

```diff
@@ -265,6 +372,129 @@ impl Editor {
+                            self.update_layout(self.width, 1.0);
+
+                            if preedit_start > 0 {
+                                self.selection = Selection::from_index(
+                                    &self.layout,
+                                    preedit_start - 1,
+                                    Affinity::Upstream,
+                                );
+                            } else {
+                                self.selection = Selection::from_index(
+                                    &self.layout,
+                                    preedit_start,
+                                    Affinity::Downstream,
+                                );
+                            }
+                        } else {
+                            self.preedit_range = Some(preedit_start..preedit_start + text.len());
+                            self.update_layout(self.width, 1.0);
+
+                            if let Some(compose_cursor) = compose_cursor {
+                                // Select the location indicated by the IME.
+                                if compose_cursor.0 == compose_cursor.1 {
+                                    self.selection = Selection::from_index(
+                                        &self.layout,
+                                        preedit_start + compose_cursor.0,
+                                        Affinity::Downstream,
+                                    );
+                                } else {
+                                    self.selection = Selection::from_cursors(
+                                        Cursor::from_index(
+                                            &self.layout,
+                                            preedit_start + compose_cursor.0,
+                                            Affinity::Downstream,
+                                        ),
+                                        Cursor::from_index(
+                                            &self.layout,
+                                            preedit_start + compose_cursor.1 - 1,
+                                            Affinity::Upstream,
+                                        ),
+                                    );
+                                }
+                            } else {
+                                // IME indicates nothing is to be selected: collapse the selection to a
+                                // caret just in front of the preedit.
+                                self.selection = Selection::from_index(
+                                    &self.layout,
+                                    preedit_start,
+                                    Affinity::Downstream,
+                                );
+                            }
+                        }
+
+                        if self.selection != old_selection {
+                            self.set_ime_cursor_area(window);
+                        }
```

Comment:
A potential issue I see with this is if for some reason `self.selection == old_selection`, but we have not sent the area yet, the platform is unaware of the area.

Maybe we can unconditionally send the area in `Ime::Enabled`.

Reply:
(In the masonry case, this would have to calculate window coordinates of the selection)